lib/modules/datasource/metadata.ts

```diff
@@ -170,3 +182,17 @@ export function addMetaData(
     }
   }
 }
+
+// delete homepage if its the same as the sourceUrl
+function shouldDeleteHomepage(dep: ReleaseResult): boolean {
+  if (dep.homepage === undefined) {
+    return false;
+  }
+  if (isGitHubUrl(dep.homepage)) {
+    return dep.sourceUrl === massageGithubUrl(dep.homepage);
+  }
+  if (isGitLabUrl(dep.homepage)) {
+    return dep.sourceUrl === massageGitlabUrl(dep.homepage);
+  }
+  return dep.sourceUrl === dep.homepage;
```

Comment:
thinking again about this. We should not remove homepage when it's not a valid gitlab or github url.
We need `sourceUrl` for changelogs. so it doesn't matter when they are equal but don't point to a supported changelog source

check also other comment about `detectPlatform`

Reply:
I've made a check just for `gitlab` and `github` urls.